credly/views.py

```diff
@@ -0,0 +1,599 @@
+import base64
+import json
+
+from datetime import date, timedelta
+
+import requests;
+from requests.auth import AuthBase
+
+import slumber
+from slumber.exceptions import HttpClientError, HttpNotFoundError
+from django.conf import settings
+
+from django.http import HttpResponse
+from django.shortcuts import get_object_or_404
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_GET, require_http_methods
+
+from teach.settings import CREDLY_API_KEY, CREDLY_APP_SECRET, CREDLY_API_URL, CREDLY_ACCOUNT_EMAIL, CREDLY_ACCOUNT_PASSWORD, MozillaAccountId
+from credly.models import UserCredlyProfile, create_new_user, save_user_token, get_credly_access_token, get_credly_token_age
+
+# ------------------------------------------------------
+# Auth handler to ensure headers/params for Credly calls
+# ------------------------------------------------------
+
+class ApiAuth(AuthBase):
+    def __init__(self, api_key, api_secret, email=None, password=None):
+        self.api_key = api_key
+        self.api_secret = api_secret
+
+        self.email = email
+        self.password = password
+
+    def __call__(self, r):
+        r.headers['X-Api-Key'] = self.api_key
+        r.headers['X-Api-Secret'] = self.api_secret
+
+        if self.email and self.password:
+            base64string = base64.encodestring('%s:%s' % (self.email, self.password))[:-1]
+            auth_header = "Basic %s" % base64string
+            r.headers['Authorization'] = auth_header
+
+        #print r.headers
+        return r
+
+
+# ------------------------------------------------------
+#             Requests helper definitions
+# ------------------------------------------------------
+
+def XAPIHeaders():
+    return {
+        'X-Api-Key': CREDLY_API_KEY,
+        'X-Api-Secret': CREDLY_APP_SECRET
+    }
+
+def credlyAuth():
+    return ApiAuth(
+        CREDLY_API_KEY,
+        CREDLY_APP_SECRET,
+        email = CREDLY_ACCOUNT_EMAIL,
+        password = CREDLY_ACCOUNT_PASSWORD
+    )
+
+def APICredentials():
+    return {
+        'email': CREDLY_ACCOUNT_EMAIL,
+        'password': CREDLY_ACCOUNT_PASSWORD
+    }
+
+# ------------------------------------------------------
+#               Route handling helpers
+# ------------------------------------------------------
+
+'''
+  FIXME: TODO: this is repeated from teach/views.py -> refactor
+'''
+def form_http_response(request, status_override=200):
+    origin = request.META.get('HTTP_ORIGIN')
+    valid_origins = settings.CORS_API_LOGIN_ORIGINS
```

Comment:
This might be a `cors` that was implemented previously, but there is a module that makes this much simpler : https://github.com/ottoyiu/django-cors-headers